go/apps/mixnet/queue.go

```diff
@@ -167,45 +255,68 @@ func (sq *SendQueue) dequeue() {
 		if res.c != nil {
 			// Save the connection.
 			sq.nextConn[res.id] = res.c
+		}
 
-			// Pop the message from the buffer and decrement the counter
-			// if the buffer is empty.
-			buf := sq.sendBuffer[res.id]
-			buf.Remove(buf.Front())
-			if buf.Len() == 0 {
-				sq.ct--
-			}
+		// Pop the message from the buffer and decrement the counter
+		// if the buffer is empty.
+		buf := sq.sendBuffer[res.id]
+		buf.Remove(buf.Front())
+		if buf.Len() == 0 {
+			sq.ct--
 		}
 	}
 }
 
 type senderResult struct {
-	c  net.Conn
-	id uint64
+	c   net.Conn
+	id  uint64
+	err error
```

Comment:
Passing the error through this channel is not actually necessary since the message is popped from the list in any case.